specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -364,33 +361,42 @@ export function Dialog({
     );
   }, [showIcon, defaultIcon, buttons, buttonContainer]);
 
-  const overlayElement: Props['overlayElement'] = React.useCallback(
-    (
-      props: React.ComponentPropsWithRef<'div'>,
-      contentElement: React.ReactElement
-    ) => (
-      <div
-        {...props}
-        onMouseDown={
-          closeOnOutsideClick
-            ? (event): void => {
-                // Outside click detection
-                if (
-                  modal &&
-                  typeof handleClose === 'function' &&
-                  event.target === event.currentTarget
-                ) {
-                  event.preventDefault();
-                  handleClose();
-                } else props?.onMouseDown?.(event);
-              }
-            : undefined
-        }
-      >
-        {contentElement}
-      </div>
-    ),
-    [modal, handleClose, closeOnOutsideClick]
+  const allowCloseRef = React.useRef<boolean>(false);
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      allowCloseRef.current = true;
+    }, 300);
```

Comment:
extract `300` into a variable with a comment for why such value was chosen?

i.e long enough to prevent accidental dialog close during the brief moment of dialog resize, but not long enough to falsely intercept many intentional clicks